provider/defaultingress/hcp/resource.go

```diff
@@ -0,0 +1,330 @@
+}
+
+func (r *DefaultIngressResource) populateDefaultIngress(
+	ctx context.Context,
+	state *DefaultIngress) error {
+
+	var ingress *cmv1.Ingress
+	var err error
+	if common.IsStringAttributeUnknownOrEmpty(state.Id) {
+		ingress, err = r.populateDefaultIngressFromList(ctx, state)
+		if err != nil {
+			return err
+		}
+	} else {
+		ingressResp, err := r.collection.Cluster(state.Cluster.ValueString()).Ingresses().Ingress(state.Id.ValueString()).Get().SendContext(ctx)
+		if err != nil {
+			return err
+		}
+		ingress = ingressResp.Body()
+	}
+
+	return r.populateState(ingress, state)
+}
+
+func (r *DefaultIngressResource) populateDefaultIngressFromList(ctx context.Context, state *DefaultIngress) (*cmv1.Ingress, error) {
+	ingresses, err := r.collection.Cluster(state.Cluster.ValueString()).Ingresses().List().SendContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, ingress := range ingresses.Items().Slice() {
+		if ingress.Default() {
+			return ingress, nil
+		}
+	}
+
+	return nil, fmt.Errorf("failed to find default ingress")
+}
+
+func (r *DefaultIngressResource) populateState(ingress *cmv1.Ingress, state *DefaultIngress) error {
+	if state == nil {
+		state = &DefaultIngress{}
+	}
+	state.Id = types.StringValue(ingress.ID())
+	state.ListeningMethod = types.StringValue(string(ingress.Listening()))
+
+	return nil
+}
+
+func (r *DefaultIngressResource) updateIngress(ctx context.Context, state, plan *DefaultIngress,
+	clusterId string, clusterCollection *cmv1.ClustersClient) error {
+
+	if state == nil {
+		state = &DefaultIngress{Cluster: plan.Cluster}
+	}
+	// In case default ingress was not part of state till now and we want to set
+	// it we need to bring it first as we need to set specific id
+	if common.IsStringAttributeUnknownOrEmpty(state.Id) {
+		err := r.populateDefaultIngress(ctx, state)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !reflect.DeepEqual(state, plan) {
+		err := validateDefaultIngress(ctx, plan)
+		if err != nil {
+			return err
+		}
+		if plan == nil {
+			plan = &DefaultIngress{}
+		}
+
+		ingressBuilder := getDefaultIngressBuilder(ctx, state, plan)
+
+		ingress, err := ingressBuilder.Build()
+		if err != nil {
+			return err
+		}
+
+		ingressResp, err := clusterCollection.Cluster(clusterId).Ingresses().Ingress(state.Id.ValueString()).Update().
+			Body(ingress).SendContext(ctx)
+
+		if err != nil {
+			return err
+		}
+		if err := r.populateState(ingressResp.Body(), plan); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getDefaultIngressBuilder(ctx context.Context, state, plan *DefaultIngress) *cmv1.IngressBuilder {
+	ingressBuilder := cmv1.NewIngress()
+	if !common.IsStringAttributeUnknownOrEmpty(plan.ListeningMethod) && state.ListeningMethod != plan.ListeningMethod {
+		ingressBuilder.Listening(cmv1.ListeningMethod(plan.ListeningMethod.ValueString()))
+	}
+	return ingressBuilder
+}
+
+func validateDefaultIngress(ctx context.Context, state *DefaultIngress) error {
+	return nil
```

Comment:
Is this intended?

Reply:
Yes, for now. The validation for classic is around attributes that HCP doesn't have. I can just remove it as it's a no op, but it's mostly for someone in the future to implement...

Reply:
Ok. I suggest adding a comment explaining what you just said here then